adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java

```diff
@@ -226,12 +244,28 @@ protected void validate() {
       properties.put("fs.adl.impl", "org.apache.hadoop.fs.adl.AdlFileSystem");
       properties.put("fs.AbstractFileSystem.adl.impl", "org.apache.hadoop.fs.adl.Adl");
       properties.put("dfs.adls.oauth2.access.token.provider.type", "ClientCredential");
-      properties.put("dfs.adls.oauth2.refresh.url", refreshTokenURL);
-      properties.put("dfs.adls.oauth2.client.id", clientId);
-      properties.put("dfs.adls.oauth2.credential", credentials);
+      if (keyVaultUrl != null && !keyVaultUrl.isEmpty()) {
+        Map<String, String> credentials = AzureClientSecretService.getADLSSecretsUsingJceksAndKV(keyVaultUrl, getKvKeyNamesMap(kvKeyNames), properties);
+        properties.put("dfs.adls.oauth2.refresh.url", credentials.get("Refresh Token Url Key-Name"));
```

Comment:
This looks like the plugin is expecting a key named 'Refresh Token Url Key-Name' in the kvKeyNamesMap. This is different than what is documented in the reference doc and is a difficult contract for a user to understand. I could not understand what was expected until reading this code.

Reply:
It seems like the correct thing to do for this is actually to create a secure store integration such that the existing ${secure(key-name)} macro would lookup the value from KeyVault. This involves writing a secure store extension for CDAP. See https://github.com/cdapio/cdap/tree/develop/cdap-securestore-ext-cloudkms for an example of one that uses Google Cloud KMS. This would allow admins to setup CDAP to read from KeyVault for any plugin, not just special casing this one. Right now this plugin documentation contains information for cluster admins about various config files that would be confusing for normal users and irrelevant with cloud profiles.

If that is not possible for some reason, a better approach would be to introduce a single new boolean property 'Read Credentials From KeyVault'. When true, it uses whatever is given for the credential as the key to the KeyVault. That way there is no ambiguity in terms of what is required.

Do the refresh url and client id need to be secure? Isn't it enough for the credential to be secure?